main/autogen/autoloader.cc

```diff
@@ -222,6 +223,25 @@ string DefTree::renderAutoloadSrc(const core::GlobalState &gs, const AutoloaderC
             }
             fmt::format_to(std::back_inserter(buf), "}})\n", fullName);
         }
+
+        if (!alCfg.packagedShimPaths.empty() && gs.packageDB().isPackage(qname.nameParts)) {
```

Comment:
While this code runs on every constant, `isPackage` is quite fast due to binary search and hence total overhead on Stripe codebase is not much.